awscli/customizations/emr/createdefaultroles.py

```diff
@@ -215,14 +215,15 @@ def _construct_role_and_role_policy_structure(
             list.append({'Role': response['Role'], 'RolePolicy': policy})
             return list
 
-    def _check_if_role_exists(self, role_name, parsed_globals):
```

Comment:
The issue is that the command is exceedingly difficult to test. It needs to have some significant refactoring. At the moment though, we have a broken command.